app/src/main/java/lt/vilnius/tvarkau/fragments/ReportTypeListFragment.kt

```diff
@@ -0,0 +1,92 @@
+package lt.vilnius.tvarkau.fragments
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.Toast
+import com.vinted.extensions.gone
+import com.vinted.extensions.visible
+import kotlinx.android.synthetic.main.app_bar.*
+import kotlinx.android.synthetic.main.fragment_report_type_list.*
+import lt.vilnius.tvarkau.BaseActivity
+import lt.vilnius.tvarkau.R
+import lt.vilnius.tvarkau.activity.ReportRegistrationActivity
+import lt.vilnius.tvarkau.backend.ApiMethod
+import lt.vilnius.tvarkau.backend.ApiRequest
+import lt.vilnius.tvarkau.backend.GetProblemTypesParams
+import lt.vilnius.tvarkau.views.adapters.ReportTypesListAdapter
+import rx.Subscription
+import timber.log.Timber
+
+/**
+ * @author Martynas Jurkus
+ */
+class ReportTypeListFragment : BaseFragment(), ReportTypesListAdapter.ReportTypeSelectedListener {
+
+    private var subscription: Subscription? = null
+    private var reportTypesListAdapter: ReportTypesListAdapter? = null
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
+        return inflater.inflate(R.layout.fragment_report_type_list, container, false)
+    }
+
+    override fun onViewCreated(view: View?, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        with(activity as BaseActivity) {
+            setSupportActionBar(toolbar)
+
+            supportActionBar?.let {
+                it.setDisplayHomeAsUpEnabled(true)
+                it.setTitle(R.string.title_choose_problem_type)
+            }
+        }
+        toolbar.setOnMenuItemClickListener {
+            when (it.itemId) {
+                android.R.id.home -> {
+                    activity.onBackPressed()
+                    true
+                }
+                else -> false
+            }
+        }
+    }
+
+    override fun onActivityCreated(savedInstanceState: Bundle?) {
+        super.onActivityCreated(savedInstanceState)
+
+        val request = ApiRequest<GetProblemTypesParams>(ApiMethod.GET_PROBLEM_TYPES, null)
+
+        legacyApiService.getProblemTypes(request)
+                .toSingle()
+                .doOnSubscribe { report_types_progress.visible() }
```

Comment:
Can we have ID naming convention changed from snake to camel case? So this would look like `reportTypesProgress` instead of `report_types_progress`?